R/gather.R

```diff
@@ -14,9 +14,26 @@
 #'   [rlang::quo_name()] (note that this kind of interface where
 #'   symbols do not represent actual objects is now discouraged in the
 #'   tidyverse; we support it here for backward compatibility).
-#' @param ... Specification of columns to gather. Use bare variable names.
-#'   Select all variables between x and z with `x:z`, exclude y with
-#'   `-y`. For more options, see the [dplyr::select()] documentation.
+#' @param ... A selection of columns. These arguments are passed to
+#'   [tidyselect::vars_select()] and are treated specially for
+#'   selection:
+#'
+#'     * Calls and complex expressions are evaluated in the context.
```

Comment:
This gets too technical too fast. I think it would be better to keep the basics as before, and move the details to a new section.